cdap-source-control/src/test/java/io/cdap/cdap/sourcecontrol/operationrunner/RemoteSourceControlOperationRunnerTest.java

```diff
@@ -226,4 +232,80 @@ public void testRemoteSourceControlOperationRunnerAuthenticationConfigException(
 
     operationRunner.push(mockPushContext);
   }
+
+  @Test
+  public void testListSuccess() throws Exception {
+    String serverURL = gitServer.getServerURL();
+    RepositoryConfig repoConfig =
+      new RepositoryConfig.Builder(mockRepoConfig).setLink(serverURL + "ignored").build();
+    ListAppsContext testListContext = new ListAppsContext(new NamespaceId(NAMESPACE), repoConfig);
+
+    RemoteSourceControlOperationRunner operationRunner =
+      new RemoteSourceControlOperationRunner(cConf, metricsCollectionService, remoteClientFactory);
+
+    RepositoryApp app1 = new RepositoryApp(TEST_APP_NAME, getGitStyleHash(TEST_APP_SPEC));
+    Path configFile1 = Paths.get(PATH_PREFIX, app1.getName()+".json");
+    addFileToGit(configFile1, TEST_APP_SPEC, gitServer);
+
+    RepositoryApp app2 = new RepositoryApp(TEST_APP_NAME+"1", getGitStyleHash(TEST_APP_SPEC));
+    Path configFile2 = Paths.get(PATH_PREFIX, app2.getName()+".json");
+    addFileToGit(configFile2, TEST_APP_SPEC, gitServer);
+
+    RepositoryAppsResponse response = operationRunner.list(testListContext);
+    List<RepositoryApp> apps = response.getApps().stream()
+      .sorted(Comparator.comparing(RepositoryApp::getName)).collect(Collectors.toList());
+
+    Assert.assertEquals(app1, apps.get(0));
+    Assert.assertEquals(app2, apps.get(1));
+  }
+
+  @Test(expected = SourceControlException.class)
+  public void testListSourceControlException() throws Exception {
+    String serverURL = gitServer.getServerURL();
+    RepositoryConfig repoConfig =
+      new RepositoryConfig.Builder(mockRepoConfig).setLink(serverURL + "ignored").build();
+    ListAppsContext testListContext = new ListAppsContext(new NamespaceId(NAMESPACE), repoConfig);
+
+    RemoteSourceControlOperationRunner operationRunner =
+      new RemoteSourceControlOperationRunner(cConf, metricsCollectionService, remoteClientFactory);
+
+    RepositoryApp app1 = new RepositoryApp(TEST_APP_NAME, getGitStyleHash(TEST_APP_SPEC));
+    Path configFile1 = Paths.get(PATH_PREFIX, app1.getName()+".json");
```

Comment:
wonder why checkstyle didn't error out about the missing spaces around `+`

Reply:
done